### PR TITLE
[Windows] Refactor logic when window resize completes

### PR DESCRIPTION
**Pull request by loic-sharma**
Changes:

1. Moves surface buffer swapping from `FlutterWindowsView` to `CompositorOpenGL`
1. Renames `FlutterWindowsView::SwapBuffers` to `FlutterWindowsView::OnFramePresented`
2. Previously, if a resize was pending and the window was not visible Windows would unblock the platform thread before swapping buffers. This trick aimed to reduce the time the platform thread was blocked as swapping buffers previously waited until the v-blank. This logic was removed as swapping buffers no longer waits until the v-blank.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/angle_surface_manager.cc [hunk 1]
```diff
@@ -329,7 +329,7 @@ bool AngleSurfaceManager::MakeResourceCurrent() {
                          egl_resource_context_) == EGL_TRUE);
 }
 
-EGLBoolean AngleSurfaceManager::SwapBuffers() {
+bool AngleSurfaceManager::SwapBuffers() {
```
Comment (yaakovschectman) [inline shell/platform/windows/angle_surface_manager.cc:332]: What is the purpose of this change?

File: shell/platform/windows/compositor_opengl.cc [hunk 2]
```diff
@@ -154,7 +154,12 @@ bool CompositorOpenGL::Present(const FlutterLayer** layers,
                        GL_NEAREST            // filter
   );
 
-  return engine_->view()->SwapBuffers();
+  if (!engine_->surface_manager()->SwapBuffers()) {
+    return false;
+  }
+
+  engine_->view()->OnFramePresented();
+  return true;
```
Comment (yaakovschectman) [inline shell/platform/windows/compositor_opengl.cc:162]: Should this be refactored into a separate function? It looks like this and the below addition to this file are identical.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

I love it! Super since!

File: shell/platform/windows/angle_surface_manager.cc (hunk 1 above)
Comment (cbracken) [reply shell/platform/windows/angle_surface_manager.cc:332]: Anytime we can avoid leaking implementation details (including types) from behind interfaces, it's a nice plus. Users of our surface managers shouldn't need to know about the underlying usage of GL types.

File: shell/platform/windows/compositor_opengl.cc (hunk 2 above)
Comment (loic-sharma) [reply shell/platform/windows/compositor_opengl.cc:162]: They are identical, but I think the logic is short enough that it's okay to repeat. I don't feel super strongly about this though, let me know if you do and I'd be happy to update this.

Comment (yaakovschectman) [reply shell/platform/windows/compositor_opengl.cc:162]: Just a thought. I don't think it's too important either way.

**Review by yaakovschectman — APPROVED** [review]
